clio-client/src/main/scala/org/broadinstitute/clio/client/commands/Commands.scala

```diff
@@ -2,39 +2,104 @@ package org.broadinstitute.clio.client.commands
+            logger.info(
+              s"Successfully completed command $commandName." +
+                s" Response code: ${response.status}"
+            )
+          } else {
+            logger.error(
+              s"Error executing command $commandName." +
+                s" Response code: ${response.status}"
+            )
+          }
+          isSuccess
+        },
+        ClioClientConfig.responseTimeout
+      )
+    }
+  }
+  case class AddReadGroupBam()(implicit ec: ExecutionContext)
+      extends Command(commandName = "addReadGroupBam") {
+    override def execute(clioWebClient: ClioWebClient,
+                         config: BaseArgs): Boolean = {
+      //parse metadata to validate inputs
+      val json = parse(IoUtil.readMetadata(config.metadataLocation)) match {
+        case Right(value) => value
+        case Left(parsingFailure) =>
+          throw parsingFailure
+      }
 
-  def checkResponse(responseFuture: Future[HttpResponse]): Boolean = {
-    Await.result(
-      responseFuture.map[Boolean] { response =>
-        val isSuccess = response.status.isSuccess()
+      val decoded = json.as[TransferReadGroupV1Metadata] match {
+        case Right(value) => value
+        case Left(decodingFailure) =>
+          throw decodingFailure
+      }
 
-        if (isSuccess) {
-          logger.info(
-            s"Successfully completed command $name." +
-              s" Response code: ${response.status}"
+      val responseFuture =
+        clioWebClient.addReadGroupBam(
+          bearerToken = config.bearerToken,
+          transferReadGroupV1Metadata = decoded,
+          input = TransferReadGroupV1Key(
+            flowcellBarcode = config.flowcell.get,
+            lane = config.lane.get,
+            libraryName = config.libraryName.get,
+            location = Location.pathMatcher(config.location.get)
           )
-        } else {
-          logger.error(
-            s"Error executing command $name." +
-              s" Response code: ${response.status}"
-          )
-        }
-        isSuccess
-      },
-      ClioClientConfig.responseTimeout
-    )
+        )
+
+      checkResponse(responseFuture)
+    }
+  }
+
+  case class QueryReadGroupBam()(implicit ec: ExecutionContext)
+      extends Command(commandName = "queryReadGroupBam") {
+    override def execute(clioWebClient: ClioWebClient,
+                         config: BaseArgs): Boolean = {
+
+      val responseFuture = clioWebClient.queryReadGroupBam(
+        bearerToken = config.bearerToken,
+        TransferReadGroupV1QueryInput(
+          flowcellBarcode = config.flowcell,
+          lane = config.lane,
+          libraryName = config.libraryName,
+          location = config.location.map(s => Location.pathMatcher(s)),
+          lcSet = config.lcSet,
+          project = config.project,
+          sampleAlias = config.sampleAlias,
+          runDateEnd = config.runDateEnd,
+          runDateStart = config.runDateStart
+        )
+      )
+
```

Comment:
Between running and checking, I think this command should unmarshal and print the contents of `responseFuture`. Otherwise people who run queries will see a log saying it succeeded, but not the actual query response.